js/apps/system/_admin/aardvark/APP/react/src/views/visgraphs/VisNetwork.js

```diff
@@ -0,0 +1,218 @@
+import React, { useState, useEffect, useRef } from "react";
+import styled from "styled-components";
+import { Network } from "vis-network";
+import { isEqual } from "lodash";
+import { ProgressBar } from "./ProgressBar";
+
+const VisNetwork = ({graphData, graphName, options, selectedNode, onSelectNode, onSelectEdge, onDeleteNode, onDeleteEdge, onEditNode, onEditEdge, onSetStartnode, onExpandNode, onAddNodeToDb, onAddEdge}) => {
+	const [layoutOptions, setLayoutOptions] = useState(options);
+	const [contextMenu, toggleContextMenu] = useState("");
+	const [position, setPosition] = useState({ x: 10, y:10 });
+	const [contextMenuNodeID, setContextMenuNodeID] = useState();
+	const [contextMenuEdgeID, setContextMenuEdgeID] = useState();
+	const [networkData, setNetworkData] = useState();
+	
+	const [visGraphData, setVisGraphData] = useState(null);
+
+	if (!isEqual(visGraphData, graphData)) {
+		setVisGraphData(graphData);
+	};
+
+	const nodes = [
+		...graphData.nodes
+	];
+
+	const edges = [
+		...graphData.edges
+	];
+
+	const visJsRef = useRef(null);
+
+	const StyledContextComponent = styled.div`
```

Comment:
create this component outside the `VisNetwork` function. Currently this is being created in render, which can lead to bugs

Reply:
Resolved within https://github.com/arangodb/arangodb/pull/18199/commits/6b982ba1d9bdb7f16c439454f3f271d6256de79e